jasmine-standalone-3.6.0/spec/BowlingGameSpec.js

```diff
@@ -0,0 +1,198 @@
+      bowlingGame.playFrame(10);
+      bowlingGame.playFrame(5, 2);
+      expect(bowlingGame.getGameScore()).toEqual(24);
+    });
+
+    it("does not add total score of latest frame to total score of the previous frame unless the previous frame was a strike", function() {
+      bowlingGame.playFrame(7, 3);
+      bowlingGame.playFrame(5, 2);
+      expect(bowlingGame.getGameScore()).not.toEqual(24);
+    });
+
+    it("adds first throw of latest frame to total score of 2 frames previous if the 2 previous frames were both strikes", function() {
+      for (var i = 0; i < 2; i++) {
+        bowlingGame.playFrame(10);
+      }
+      bowlingGame.playFrame(5, 2);
+      expect(bowlingGame.getGameScore()).toEqual(49);
+    });
+
+    it("does not add first throw of latest frame to total score of 2 frames previous unless the 2 previous frames were both strikes", function() {
+      bowlingGame.playFrame(10);
+      bowlingGame.playFrame(5, 5);
+      bowlingGame.playFrame(5, 2);
+      expect(bowlingGame.getGameScore()).not.toEqual(49);
+    });
+
+    it("throws an error if 10 frames have already been played", function() {
+      for (var i = 0; i < 10; i++) {
+        bowlingGame.playFrame(6, 3);
+      }
+      expect(function() { bowlingGame.playFrame(6, 3); }).toThrowError(`You have already played ${bowlingGame.INDEX_OF_FINAL_FRAME + 1} frames`)
+    })
+  });
+
+  describe("areBonusRollsNeeded", function() {
+    it("returns true if the 10th frame is a spare", function() {
+      for (var i = 0; i < 9; i++) {
+        bowlingGame.playFrame(6, 3);
+      }
+      bowlingGame.playFrame(6, 4);
+      expect(bowlingGame.areBonusRollsNeeded()).toEqual(true)
+    });
+
+    it("returns true if the 10th frame is a strike", function() {
+      for (var i = 0; i < 9; i++) {
+        bowlingGame.playFrame(6, 3);
+      }
+      bowlingGame.playFrame(10);
+      expect(bowlingGame.areBonusRollsNeeded()).toEqual(true)
+    });
+
+    it("returns true if the 9th & 10th frames are strikes", function() {
+      for (var i = 0; i < 8; i++) {
+        bowlingGame.playFrame(6, 3);
+      }
+      for (var i = 0; i < 2; i++) {
+        bowlingGame.playFrame(10);
+      }
+      expect(bowlingGame.areBonusRollsNeeded()).toEqual(true)
+    });
+
+    it("returns false if the 10th frame neither a strike nor a spare", function() {
+      for (var i = 0; i < 9; i++) {
+        bowlingGame.playFrame(10);
+      }
+      bowlingGame.playFrame(6, 3);
+      expect(bowlingGame.areBonusRollsNeeded()).toEqual(false)
+    });
+  });
+
+  describe("addFinalBonusScores", function() {
+    it("if the final frame is a spare, it gives you 1 extra bonus roll and it gets added to the total game score", function() {
+      for (var i = 0; i < 9; i++) {
+        bowlingGame.playFrame(6, 3);
+      }
+      bowlingGame.playFrame(6, 4);
+      spyOn(bowlingGame,'areBonusRollsNeeded').and.returnValue(true);
```

Comment:
The Spy in this case serves the same purpose as the test set-up with playFrame. Good to get practice of them nonetheless but most useful in this challenge if you have separated out your frame and game logic classes